src/Data.mo

```diff
@@ -136,11 +140,21 @@ module {
 
     public func getMaxEligibleDeposit(threshold : Nat) : ?Entry<K> {
       for (((deposit, key), value) in state.depositsTree.entriesRev()) {
-        if (deposit <= threshold) return null;
+        if (deposit <= threshold) {
+          state.unusable_deposit.sum := state.deposit_sum;
+          state.unusable_deposit.correct := true;
+          return null;
+        };
```

Comment:
This isn't enough, or? If the queue is empty, i.e. we don't even enter the for loop once, then we also must set this to true, or?